tests/provider/google/marketing_platform/hooks/test_campaign_manager.py

```diff
@@ -0,0 +1,280 @@
+    )
+    def test_get_report(self, mock_base_hook, get_conn_mock):
+        profile_id = "PROFILE_ID"
+        report_id = "REPORT_ID"
+        file_id = "FILE_ID"
+
+        return_value = "TEST"
+        get_conn_mock.return_value.reports.return_value.files.return_value.get.\
+            return_value.execute.return_value = return_value
+
+        result = self.hook.get_report(
+            profile_id=profile_id, report_id=report_id, file_id=file_id
+        )
+
+        get_conn_mock.return_value.reports.return_value.files.return_value.get.assert_called_once_with(
+            profileId=profile_id, reportId=report_id, fileId=file_id
+        )
+
+        self.assertEqual(return_value, result)
+
+    @mock.patch(
+        "airflow.provider.google.marketing_platform.hooks."
+        "campaign_manager.GoogleCampaignManagerHook.get_conn"
+    )
+    @mock.patch(
+        "airflow.provider.google.marketing_platform.hooks."
+        "campaign_manager.GoogleCloudBaseHook.__init__"
+    )
+    def test_get_report_file(self, mock_base_hook, get_conn_mock):
+        profile_id = "PROFILE_ID"
+        report_id = "REPORT_ID"
+        file_id = "FILE_ID"
+
+        return_value = "TEST"
+        get_conn_mock.return_value.reports.return_value.files.return_value.get_media.return_value = (
+            return_value
+        )
+
+        result = self.hook.get_report_file(
+            profile_id=profile_id, report_id=report_id, file_id=file_id
+        )
+
+        get_conn_mock.return_value.reports.return_value.files.return_value.get_media.assert_called_once_with(
+            profileId=profile_id, reportId=report_id, fileId=file_id
+        )
+
+        self.assertEqual(return_value, result)
+
+    @mock.patch(
+        "airflow.provider.google.marketing_platform.hooks."
+        "campaign_manager.GoogleCampaignManagerHook.get_conn"
+    )
+    @mock.patch(
+        "airflow.provider.google.marketing_platform.hooks."
+        "campaign_manager.GoogleCloudBaseHook.__init__"
+    )
+    def test_insert_report(self, mock_base_hook, get_conn_mock):
+        profile_id = "PROFILE_ID"
+        report = {"body": "test"}
+
+        return_value = "TEST"
+        get_conn_mock.return_value.reports.return_value.insert.return_value.execute.return_value = (
+            return_value
+        )
+
+        result = self.hook.insert_report(profile_id=profile_id, report=report)
+
+        get_conn_mock.return_value.reports.return_value.insert.assert_called_once_with(
+            profileId=profile_id, body=report
+        )
+
+        self.assertEqual(return_value, result)
+
+    @mock.patch(
+        "airflow.provider.google.marketing_platform.hooks."
+        "campaign_manager.GoogleCampaignManagerHook.get_conn"
+    )
+    @mock.patch(
+        "airflow.provider.google.marketing_platform.hooks."
+        "campaign_manager.GoogleCloudBaseHook.__init__"
+    )
+    def test_list_reports(self, mock_base_hook, get_conn_mock):
```

Comment:
Can you add tests for list with multiple pages?

Reply:
Done 🐍